### PR TITLE
use converters for retrofit gen

### LINKED ISSUE #37: Protobuf support
At least that what's I understand from this line https://github.com/Jaguar-dart/client/blob/master/resty/lib/src/response.dart#L514 
Am I right @tejainece ? I tried to put my proto repo for serialization but I get this error:
```
Unhandled exception:
FormatException: Unexpected character (at line 2, character 1)
name
^

#0      _ChunkedJsonParser.fail (dart:convert/runtime/libconvert_patch.dart:1358:5)
#1      _ChunkedJsonParser.parseNumber (dart:convert/runtime/libconvert_patch.dart:1254:9)
#2      _ChunkedJsonParser.parse (dart:convert/runtime/libconvert_patch.dart:922:22)
#3      _parseJson (dart:convert/runtime/libconvert_patch.dart:29:10)
#4      JsonDecoder.convert (dart:convert/json.dart:540:36)
#5      JsonCodec.decode (dart:convert/json.dart:167:41)
#6      StringResponse.decode (package:jaguar_resty/src/response.dart:514:26)
#7      Get.one (package:jaguar_resty/src/jaguar_resty_base.dart:387:70)
<asynchronous suspension>
#8      _$Api.getPerson (package:client/main.dart:30:16)
<asynchronous suspension>
#9      MyApi.getPerson (package:client/main.dart:20:18)
#10     main (package:client/main.dart:43:28)
<asynchronous suspension>
#11     _startIsolate.<anonymous closure> (dart:isolate/runtime/libisolate_patch.dart:289:19)
#12     _RawReceivePortImpl._handleMessage (dart:isolate/runtime/libisolate_patch.dart:171:12)

```

### PR DESCRIPTION
**Pull request by jaumard**
fix #37

### CODE REVIEW

**Comment by jaumard** [comment]
@tejainece can I remove totally the `jsonConverter` ? because now that we have converters it's kind of weird to have 2 ways of doing serialization...

File: retrofit/lib/client/client.dart [hunk 1]
```diff
@@ -6,45 +6,32 @@ import 'package:jaguar_resty/jaguar_resty.dart';
 
 abstract class ApiClient {
   Route get base;
-  JsonRepo get jsonConverter;
+  static const contentTypeJson = 'application/json';
```
Comment (tejainece) [inline retrofit/lib/client/client.dart]: It is not good to declare them here. I think the best way to get these constants is `dart:io`'s ContentType class. If that is not good enough, we can create a `ContentType` class and put it in resty.

Comment (jaumard) [reply retrofit/lib/client/client.dart]: ho didn't know it was available on `dart:io` ! I'll change to use those instead then ! much better thanks !

Comment (jaumard) [reply retrofit/lib/client/client.dart]: :( unfortunately we can't reuse the `dart:io ` one :( annotation can only call `const` constructor and `ContentType` is not const :( so I'll create a class in resty for that

Comment (jaumard) [reply retrofit/lib/client/client.dart]: Done and pushed @tejainece :) much better like this indeed !

**Comment by jaumard** [comment]
I add to change some pub spec to use local resty. I let you revert those when you publish the libs.

**Comment by tejainece** [comment]
Made some minor changes and published them as `2.8.x`.

**Comment by jaumard** [comment]
can't use it :( is it possible to put >= 2.0.0 ? 
```
The current Dart SDK version is 2.1.0-dev.9.4.flutter-f9ebf21297.

Because jaguar_retrofit >=2.8.1 depends on jaguar_mimetype any which requires SDK version >=2.1.0 <3.0.0, jaguar_retrofit >=2.8.1 is forbidden.
So, because client depends on jaguar_retrofit ^2.8.1, version solving failed.

```

**Comment by tejainece** [comment]
Fixed it.

**Pull request merged**